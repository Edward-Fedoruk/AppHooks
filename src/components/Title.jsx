import React from "react"
import { withStyles } from "@material-ui/core/styles"
import Typography from "@material-ui/core/Typography"
import PropTypes from "prop-types"

const titleStyles = ({ breakpoints }) => ({
  title: {
    flexGrow: 1,
    fontSize: "25px",
    textTransform: "capitalize",
    color: "#192B7F",
    fontWeight: "bold",

    [breakpoints.down(425)]: {
      fontSize: "18px",
    },
  },
})

const Title = ({ classes, styles, children }) => (
  <Typography
    variant="h2"
    color="primary"
    className={classes.title}
    style={styles}
  >
    { children }
  </Typography>
)

Title.defaultProps = {
  styles: {},
}

Title.propTypes = {
  classes: PropTypes.object.isRequired,
  styles: PropTypes.object,
  children: PropTypes.string.isRequired,
}

export default withStyles(titleStyles)(Title)
