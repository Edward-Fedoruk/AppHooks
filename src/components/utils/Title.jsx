import React from "react"
import { withStyles } from "@material-ui/core/styles"
import Typography from "@material-ui/core/Typography"
import PropTypes from "prop-types"

const titleStyles = ({ breakpoints }) => ({
  title: {
    flexGrow: 1,
    fontSize: "25px",
    textTransform: "capitalize",
    color: "#192B7F",
    fontWeight: "400",

    [breakpoints.down(425)]: {
      fontSize: "18px",
    },
  },
})

const Title = ({ classes, styles, children }) => (
  <Typography
    variant="h2"
    color="primary"
    className={classes.title}
    style={styles}
  >
    { children }
  </Typography>
)

Title.defaultProps = {
  styles: {},
  children: "",
}

Title.propTypes = {
  classes: PropTypes.object.isRequired,
  styles: PropTypes.object,
  children: PropTypes.string,
}

export default withStyles(titleStyles)(Title)
