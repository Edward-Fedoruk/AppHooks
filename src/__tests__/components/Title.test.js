import "jsdom-global/register"
import React from "react"
import { shallow, mount } from "enzyme"
import Title from "../../components/utils/Title"

describe("Title", () => {
  it("should match snapshot", () => {
    const tree = shallow(
      <Title>test</Title>
    )
    expect(tree).toMatchSnapshot()
  })
  it("render text", () => {
    const wrapper = mount(
      <Title>test</Title>
    )

    expect(wrapper.text()).toBe("test")
  })
})
