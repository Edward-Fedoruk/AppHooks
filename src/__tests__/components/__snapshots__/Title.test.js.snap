// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should match snapshot 1`] = `
<Title
  classes={
    Object {
      "title": "Title-title-1",
    }
  }
  styles={Object {}}
>
  test
</Title>
`;
